geode-redis/src/main/java/org/apache/geode/redis/internal/executor/hash/GeodeRedisHashSynchronized.java

```diff
@@ -89,6 +89,11 @@ public int hdel(List<ByteArrayWrapper> subList) {
       numDeleted.set(oldHash.size() - newHash.size());
       return newHash;
     });
+
+    if (hgetall().size() == 0) {
```

Comment:
Instead of ".size() == 0" I think ".isEmpty()" is better.

Reply:
Calling "hgetall" and then checking the size does not seem thread safe. Couldn't another  thread add to this set and find the existing one we emptied out and add to it right after this thread checks the size and then we will unregister a non-empty hash?
If we do have a concurrency issue I'm still okay with this change since it fixes the non-concurrent case. Maybe we could just add a TODO comment that concurrency needs to be addressed?

Reply:
True, I will update to .isEmpty()

Reply:
Yeah, I think we just put in the non-concurrent case for now since we are still deciding on an approach (compound key vs. function/delta vs. something else).  I will add a TODO, though.